Use the Terms UI to manage BP Types

imath opened the pull request:
Hi!

I know this wasn't the direction that was chosen for the BP Types UI, but I believe since this [comment](https://buddypress.trac.wordpress.org/ticket/7181#comment:10) was made, things have evolved.

1. As BuddyPress 7.0.0 will/could require WordPress 4.9 : term meta can now be used to store specific BP Types data.
2. We can keep on using a global registry for BP Types and use the WordPress Terms UI
3. Using a Post type to do something else than writing/publishing content seems weird to me. Moreover when we're talking about an object "Term" that should organize posts :)

I strongly believe the best way to maintain the BP Types UI in the long run is to use something that was primarly designed to manage terms > WP Terms UI.

In this PR, I'm adapting this WP Terms UI to match BP Types need:
- Ajax actions and bulk actions are neutralized,
- Quick edit row action is neutralized,
- Add/edit Forms and WP List Terms Table Columns are customized, 
- Only the BP Types needed data are editable.

Here's a screenshot of the BP Member Types UI:

![member-types-manage](https://user-images.githubusercontent.com/1834524/85568272-d4ea5080-b631-11ea-87ca-014e4ca4ced2.png)

Here's a screenshot of the BP Group Types UI:

![group-types-manage](https://user-images.githubusercontent.com/1834524/85568486-0236fe80-b632-11ea-9888-011e857361c2.png)

**NB**

- BP Types registered by code cannot be deleted.
- Most BP Types registered by code data can be overridden by edits saved into the database (only slug and name cannot be edited).

It works fine on WordPress 5.4 + BuddyPress 7.0-alpha when WordPress is configured as a single site. Just like it was the case for the Post type scenario, some adjustments are needed when BuddyPress is network activated as `edit.php`, `post.php`, `edit-tags.php` and `edit-tag-form.php` are not available into the network admin.

imath committed:
Use the Terms UI to manage BP Types

boonebgorges commented on includes/functions.php:
```diff
@@ -0,0 +1,415 @@
+		'bp_group_type'               => 'bp_group',
+	);
+
+	foreach ( $bp_type_taxonomies as $bp_type_taxonomy ) {
```
Is this necessary just to enable 'show_ui'?

imath replied:
Actually: no 😄 . We don't need to unregister/reregister taxonomies. Your question made me look deeper into WordPress and the `register_taxonomy_args` should be used to extend the BP Type taxonomy arguments. That's what I've done in https://github.com/buddypress/bp-types-ui/pull/6/commits/798f2af6960730297ada75918261182e43c3388f

boonebgorges replied:
Nice!

boonebgorges reviewed (commented):
Thanks for working on this, @imath. I agree that this approach makes more sense than using a CPT, or the CPT UI in the Dashboard.

For navigation in the Network Admin, I think that submenu items under Groups and Users, which point to the relevant edit-tags.php page on the root blog, is probably fine. Not ideal - especially in the case of Groups, where you wouldn't have a top-level menu item to highlight as "current" on the root blog Dashboard - but maybe good enough for our purposes here.

imath committed:
Use the `register_taxonomy_args` filter to extend BP taxonomy args

Unregistering - Reregistering the BP taxonomies is actually not needed thanks to this filter.

boonebgorges commented:
This looks pretty good to me, @imath - feels like it can be merged, except for the fact that there's no way to get to the screen if you're on Multisite. What do you think about that?

imath commented:
Thanks for your feedback @boonebgorges. I think we could try to use a trick for multisite configs where BuddyPress is network activated:
- do as you suggested into network admin adding sub-menus for BP Types which point to the relevant edit-tags page of the root blog admin.
- into the root blog admin add a bp-groups menu which point to the network Groups screen and use it as the parent for the Group Types screen

I'll test it on a multisite config to see how it goes 😉

imath committed:
Handle BP Types admin menu on multisite WP configs

- Adds a users.php submenu in network admin to go to the root blog Member types screen.
- Adds a bp-groups submenu in network admin to go to the root blog Group types screen.
- Adds a bp-groups menu to the root blog menu when BuddyPress is network activated.
- Adds a redirection to the Groups Network Admin screen when BuddyPress is network activated.

imath commented:
@boonebgorges in https://github.com/buddypress/bp-types-ui/pull/6/commits/f014b29ab6f6990799e5702eb38b0d3ec16f62f2 I'm applying what I've described in my previous comment. This makes sure the admin can access to the BP Type Manage screens from the Network admin.

boonebgorges commented:
Thank you!
